src/client/Microsoft.Identity.Client/PlatformsCommon/DeviceAuthHelper.cs

```diff
@@ -74,14 +78,34 @@ public static string GetBypassChallengeResponse(Dictionary<string, string> respo
 
         public static bool CanOSPerformPKeyAuth()
         {
+            if (s_canOSPerformPKeyAuth != null)
+            {
+                return (bool)s_canOSPerformPKeyAuth;
+            }
+
             //PKeyAuth can only be performed on operating systems with a major OS version of 6. 
             //This corresponds to windows 7, 8, 8.1 and their server equivilents.
-#if DESKTOP
-            if (Environment.OSVersion.Version.Major == 6)
+            //Environment.OSVersion as it will return incorrect information on some operating systems
+#if NET_CORE
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                && !RuntimeInformation.OSDescription.Contains("Windows 10"))
+            {
+                s_canOSPerformPKeyAuth = true;
+                return true;
+            }
+
+#elif DESKTOP
+            var reg = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows NT\CurrentVersion");
```

Comment:
Is there a blog post or stackoverflow thread that advises this approach? We should add a link to it as a comment